addons/event_sale/tests/test_event_sale.py

```diff
@@ -121,3 +129,23 @@ def test_01_ticket_price_with_pricelist_and_tax(self):
         })
         sol.product_id_change()
         self.assertEqual(so.amount_total, 660.0, "Ticket is $1000 but the event product is on a pricelist 10 -> 6. So, $600 + a 10% tax.")
+
+    @users('salesperson')
+    def test_unlink_so(self):
+        """ This test ensures that when deleting a sale order, if the latter is linked to an event registration,
+        the number of expected seats will be correctly updated """
+        event = self.register_person.event_registration_ids.event_id
```

Comment:
I would rebrowse explicitely. In 13 or 14+ (don't remember) due to cache you may loose some ACLs issue if you don't explicitly browse. Like ``event = self.env['event.event'].browse(self.event.ids)`` <- if we store the test event on self, which could be interesting to ease test reading